ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -378,6 +378,7 @@ public void replicate() {
     logger.trace("Thread name: {} Start RemoteReplicaGroup replication", threadName);
     List<RemoteReplicaGroup> remoteReplicaGroups = new ArrayList<>();
     int remoteReplicaGroupId = 0;
+    Map<Integer, Long> groupFinishTime = new HashMap<>();
```

Comment:
minor comment: In general, consider moving away from maps and think in terms of objects. Maps are useful for competitive programming but objects are the norm in shared code bases for several reasons. I am not too particular about it in this case.

For example, this metric could have been added as follows.

```
class RemoteReplicaGroup {
   long idleTime = 0;
   public void incIdleTime(long delta) {...}
   public long getIdleTime() {...}
}
```

Reply:
Added like this.
Will try to follow object oriented design and not rely on individual data structures from now on.